ERCS/erc-7662.md

```diff
@@ -0,0 +1,273 @@
+
+## Motivation
+
+The creation and trading of AI Agent NFTs are a natural fit and offer the potential for an entirely new onchain market. This requires some custom data to be embedded in the NFT through a custom struct and this needs to be standardized so that any marketplace or AI Agent management product, among others, know how to create and parse AI Agent NFTs.  
+
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+
+All ERC-XXXX compliant contracts MUST implement the standard [ERC-721](./eip-721.md) functionality for minting and transferring NFTs, and MUST additionally implement this standard's Agent struct
+
+```solidity
+   
+struct Agent {
+  string name;
+  string description;
+  string model;
+  string userPromptURI;
+  string systemPromptURI;
+  bool promptsEncrypted;
+}
+```
+
+and MUST implement the mapping between NFT Token ID and its Agent information.
+
+```solidity
+   
+mapping(uint => Agent) public Agents;
+```
+
+It is RECOMMENDED that this mapping is public and that the URIs for User Prompt and System Prompt are made private through encryption with decryption set to the holder of the NFT via custom contract parameters set during encryption, such as those provided by Lit Protocol. 
+
+It is conceivable to also create an implementation whereby this mapping was set to private and accessed through a custom function that restricted access to the holder of the NFT. This approach would explose the prompts through their urls though, therefore the RECOMMENDED approach is a public mapping and encryption on the URLs. This also has the benefit of publicly exposing the data in the Agent struct to verify name, description and model and that encyrpted URIs for the User Prompt and System Prompt exist. 
+
+All ERC-XXXX compliant contracts MUST implement the mintAgent function
+
+```solidity
+   
+function mintAgent(address _recipient, string memory _name, string memory _description, string memory _model, string memory _userPromptURI, string memory _systemPromptURI, string memory _imageURI, string memory _category, string memory _tokenURI) public {
+        tokenIds++;
+        bool _promptsEncrypted = false;
+        Agents[tokenIds] = Agent(_name, _description, _model, _userPromptURI, _systemPromptURI, _imageURI, _category, _promptsEncrypted);
+
+        _mint(_recipient, tokenIds);
+        collectionIds[_recipient].push(tokenIds);
+        _setTokenURI(tokenIds, _tokenURI);
+        emit AgentCreated(_name, _description, _model, _category, _recipient, tokenIds);
```

Comment:
Please do not specify the implementation of functions in an ERC. Describe the externally visible behaviour of the function, but the actual code should be left to implementers.